src/components/Events/EventsSection.tsx

```diff
@@ -19,18 +20,52 @@ const EventsSection = ({
   sectionId,
 }: EventsSectionProps) => {
   const [activeFilter, setActiveFilter] = useState({
+    id: "all",
     name: "All Events",
     description:
       "Explore our complete range of technology-focused events designed to build resilience and create innovative solutions for global challenges.",
   });
 
+  const searchParams = useSearchParams();
+
+  const categoryId = searchParams.get("categoryId");
+
+  // Effect to set active filter based on URL parameter
+  useEffect(() => {
```

Comment:
Nice, I think we should implement this pattern on the other filterable lists to have them url-addressable too, wdyt @Ushindi-Zirimwabagabo-Josue ?

Reply:
One question - if you read the category from the search params first, would it be possible to remove the local state entirely, just redirect to the new URL on filter change? Then you'll keep the URL state in sync with the UI and can keep the page entirely server rendered

Reply:
It's not super important, I think lets get this merged and we can loop back around with small improvements later